### PR TITLE
Ps ui validation

### PR DESCRIPTION
**Pull request by psantos1113**

### CODE REVIEW

File: src/main/js/store/actions/roles.js [hunk 1]
```diff
@@ -124,19 +133,20 @@ export function createNewRole(roleName) {
...
                             case 401:
-                                dispatch(saveRoleError(data.message));
+                                dispatch(saveRoleError(data));
                                 return dispatch(verifyLoginByStatus(response.status));
                             case 412:
-                                return dispatch(saveRoleError(data.message));
+                                return dispatch(saveRoleError(data));
                             default: {
-                                return dispatch(saveRoleError(data.message, null));
+                                return dispatch(saveRoleError(data));
```
Comment (gkillough) [inline src/main/js/store/actions/roles.js]: All of these just dispatch `saveRoleError()`.

File: src/main/js/store/actions/roles.js [hunk 2]
```diff
@@ -153,18 +163,25 @@ export function deleteRole(roleName) {
...
                             case 401:
-                                dispatch(deletingRoleError(data.message));
+                                dispatch(deletingRoleError(data));
                                 return dispatch(verifyLoginByStatus(response.status));
                             case 412:
-                                return dispatch(deletingRoleError(data.message));
+                                return dispatch(deletingRoleError(data));
                             default: {
-                                return dispatch(deletingRoleError(data.message, null));
+                                return dispatch(deletingRoleError(data));
```
Comment (gkillough) [inline src/main/js/store/actions/roles.js]: All of these just dispatch `deletingRoleError()`.

Comment (gkillough) [reply src/main/js/store/actions/roles.js]: Except the 401.

File: src/main/js/store/actions/users.js [hunk 3]
```diff
@@ -124,19 +134,20 @@ export function createNewUser(user) {
...
                             case 401:
-                                dispatch(saveUserError(data.message));
+                                dispatch(saveUserError(data));
                                 return dispatch(verifyLoginByStatus(response.status));
                             case 412:
-                                return dispatch(saveUserError(data.message));
+                                return dispatch(saveUserError(data));
                             default: {
-                                return dispatch(saveUserError(data.message, null));
+                                return dispatch(saveUserError(data));
```
Comment (gkillough) [inline src/main/js/store/actions/users.js]: All of these just dispatch `saveUserError()` except for the 401.

File: src/main/js/store/actions/roles.js [hunk 4]
```diff
@@ -153,18 +163,25 @@ export function deleteRole(roleName) {
...
+        }).then(() => dispatch(fetchRoles()))
+            .catch(console.error);
     };
 }
+
+export function clearRoleFieldErrors() {
+    return (dispatch) => {
+        dispatch(clearFieldErrors());
+    }
+};
```
Comment (bamandel) [inline src/main/js/store/actions/roles.js]: Not sure if this ended up being used anywhere.

Comment (bamandel) [reply src/main/js/store/actions/roles.js]: JK. the search didn't work with collapsed panels

File: src/main/js/store/actions/users.js [hunk 5]
```diff
@@ -153,18 +164,25 @@ export function deleteUser(userName) {
...
                             case 401:
-                                dispatch(deletingUserError(data.message));
+                                dispatch(deletingUserError(data));
                                 return dispatch(verifyLoginByStatus(response.status));
                             case 412:
-                                return dispatch(deletingUserError(data.message));
+                                return dispatch(deletingUserError(data));
                             default: {
-                                return dispatch(deletingUserError(data.message, null));
+                                return dispatch(deletingUserError(data));
```
Comment (jamesrichard91) [inline src/main/js/store/actions/users.js]: All of these just dispatch `deletingUserError()` except for the 401.

**Pull request merged**